spydra/src/main/java/com/spotify/spydra/submitter/api/PoolingSubmitter.java

```diff
@@ -97,21 +102,34 @@ public boolean acquireCluster(SpydraArgument arguments, DataprocAPI dataprocAPI)
     }
 
     // All has failed us - try to create a cluster the good 'ol fashioned way.
+    return newPooledCluster(arguments, dataprocAPI);
+  }
+
+  private boolean newPooledCluster(SpydraArgument arguments, DataprocAPI dataprocAPI)
+      throws IOException {
+    // Label the pooled cluster with the client id. Unknown client ids all end up in their own pool.
+    arguments.addOption(arguments.cluster.options, SpydraArgument.OPTION_LABELS,
+        POOLED_CLUSTER_CLIENTID_LABEL + "=" + arguments.getClientId());
     return super.acquireCluster(arguments, dataprocAPI);
   }
 
+
   @Override
   public boolean releaseCluster(SpydraArgument arguments, DataprocAPI dataprocAPI)
       throws IOException {
     boolean shouldCollect;
     // If we're pooling clusters the cluster will live long enough for history to be collected
     shouldCollect = !arguments.isPoolingEnabled();
 
-    // Check if the cluster is healthy, if it's not it might not be able to self-destruct.
-    shouldCollect = shouldCollect || dataprocAPI.listClusters(arguments).stream()
-        .filter(cluster -> cluster.clusterName.equals(arguments.getCluster().getName()))
-        .findAny().map(cluster -> cluster.status.state.equals(Cluster.Status.ERROR)).orElse(false);
-
+    Map<String, String> clusterFilter = ImmutableMap.of(
+            "status.state", "ERROR",
+            OPTIONS_FILTER_LABEL_PREFIX + SPYDRA_CLUSTER_LABEL, "",
+            OPTIONS_FILTER_LABEL_PREFIX + POOLED_CLUSTER_CLIENTID_LABEL, arguments.getClientId()
+    );
+    shouldCollect = shouldCollect || dataprocAPI.listClusters(arguments, clusterFilter).stream()
+            .filter(cluster -> cluster.clusterName.equals(arguments.getCluster().getName()))
```

Comment:
Instead of this filter, use `clusterName` in the clusterFilter.

Reply:
Sth like this?
``` "cluster.clusterName", arguments.getCluster().getName(), ```